style_guide.rst

```diff
@@ -0,0 +1,91 @@
+Style Guide for APIRUS related projects.
+########################################
+
+This document is designed to serve as a style guide for the APIRUS related pojects. The goal is to have a consistent coding style among the inter-related projects.
+
+VERSION: 0.1
+
+Overall Style
+=============
+
+Follow:
+
+ - PEP 0008 -- Style Guide for Python Code
+
+and
+
+ - PEP 0257 -- Docstring Conventions
+
+But there are more details that are unspecified in those documents, so those are clarified here.
+
+Quote style
+-----------
+
+single vs double quotes? -- someone decide, I don't care :-)
+
+
+Specifying parameters to functions in docstrings
+------------------------------------------------
+
+I think we should use either the "raw" sphinx approach::
+
+  :param name: the value you want to pass in
+  :type name: the type required for name
+
+Or adopt the numpy extensions -- they are nicer, but require an extra plug-in...
+
+https://pypi.python.org/pypi/numpydoc
+
+
+Line Continuations
+------------------
+
+long lines need to get split, but there are multiple ways to do it.
+
+Use un-closed brackets whenever possible to support continuation.
+
+If a function definition or function call has too many parmaeters to all fit on one line, then put each on on it's own line. i.e.
+
+::
+
+    def __init__(self,
+                 node_lon=None,
+                 node_lat=None,
+                 center_lon=None,
+                 center_lat=None,
+                 edges=None,
+                 node_padding=None,
+                 edge1_padding=None,
+                 edge2_padding=None,
+                 grid_topology_var=None,
+                 variables=None,
+                 grid_variables=None,
+                 dimensions=None,
+                 node_dimensions=None,
+                 node_coordinates=None,
+                 edge1_coordinates=None,
+                 edge2_coordinates=None,
+                 angles=None,
+                 )
```

Comment:
Ironically, I truncated it -- there are quite a few more ;-)

Though I don't have a problem with an initializer with all the (optional)
parameters it needs to fully initialize the object.

And there are other cases where a lot of optional parameters make sense.

In any case, too much to fit on one line is not rare -- hence the style
suggestion.



On Thursday, March 10, 2016, Filipe <notifications@github.com> wrote:

> In style_guide.rst
> <https://github.com/ioos/APIRUS/pull/18#discussion_r55664711>:
>
> > +                 center_lon=None,
> > +                 center_lat=None,
> > +                 edges=None,
> > +                 node_padding=None,
> > +                 edge1_padding=None,
> > +                 edge2_padding=None,
> > +                 grid_topology_var=None,
> > +                 variables=None,
> > +                 grid_variables=None,
> > +                 dimensions=None,
> > +                 node_dimensions=None,
> > +                 node_coordinates=None,
> > +                 edge1_coordinates=None,
> > +                 edge2_coordinates=None,
> > +                 angles=None,
> > +                 )
>
> @rhattersley <https://github.com/rhattersley> has a point. (And that
> discussion should be in pysgrid as this was copied from there.)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ioos/APIRUS/pull/18/files#r55664711>.
>


-- 
Christopher Barker, PhD

Python Language Consulting
  - Teaching
  - Scientific Software Development
  - Desktop GUI and Web Development
  - wxPython, numpy, scipy, Cython